include/GameSpecificComponents.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "Prelude.h"
+
+#define SCREEN_WIDTH 800
+#define SCREEN_HEIGHT 600
+#define SPEED_MOD 300.0f
+#define GRAVITY_MOD 400.0f
+#define JUMP_MOD 600.0f
+
+struct Player {
+	bool is_grounded;
```

Comment:
Please use Definitions.h to be more specific about your types. This is useful as both an exercise for when you get a job in a big company and it has its own type-definitions, and as a real-world exercise in precision under stress. In these cases, rewrite this as `Bool`, `F32` and `F64` (for double). Below, there's `const char*` which is `LiteralString`. 

Pointers serve several functions in C++, so we want to be really specific about our intent: we can have a `RawPtr<T>` if we want a real pointer to something. We can use `SharedPtr` for shared ownership (most cases), `UniquePtr` for unique ownership, and `StaticArray<T, N>` to define a static array of type T and size N, a `DynamicArray<T>` to define a vector, and a `SmallBuffer<T>`  for when you know you have like... 10 elements; also `Ref<T>` for `T&`, and a lot of other things. All of this is in Definitions.h, use it often and get used to it!

Reply:
Also, our code is written in the ANSI C++ style, and not the Kernighan & Ritchie style, meaning that `{}` go on their own separate lines.